### PR TITLE
remove redundant error test case in autoscaling validation

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: allencloud <allen.sun@daocloud.io>

**What this PR does / why we need it**:
I think the error test case I removed is the same as https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/autoscaling/validation/validation_test.go#L393-L412 .
So maybe removal of this part is proper, since it would reduce the codes and save a little bit time in testing.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
NONE

**Special notes for your reviewer**:
NONE

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/autoscaling/validation/validation_test.go [hunk 1]
```diff
@@ -328,32 +328,6 @@ func TestValidateHorizontalPodAutoscaler(t *testing.T) {
...
-						Name: "myrc",
-					},
-					MinReplicas: newInt32(1),
-					MaxReplicas: 5,
-					Metrics: []autoscaling.MetricSpec{
-						{
-							Type: autoscaling.ResourceMetricSourceType,
-							Resource: &autoscaling.ResourceMetricSource{
-								Name: api.ResourceCPU,
-								TargetAverageUtilization: newInt32(-70),
```
Comment (sttts) [inline pkg/apis/autoscaling/validation/validation_test.go]: Redundant to which other test case?

Comment (sttts) [inline pkg/apis/autoscaling/validation/validation_test.go]: ic: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/autoscaling/validation/validation_test.go#L393-L412 .

**Comment by allencloud** [comment]
/assign @smarterclayton
Could you help to review this? Thanks a lot.

**Comment by mbohlool** [comment]
I am not that familiar with this e2e tests, but I noticed that those two tests have different `TargetAverageUtilization` values. Do they still considered to be duplicate with different `TargetAverageUtilization` values?

**Comment by allencloud** [comment]
@mbohlool 
Thanks for your feedback. I admit that TargetAverageUtilization has different values.
While in the validation code there is code like following, https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/autoscaling/validation/validation.go#L211-L213:
```
if src.TargetAverageUtilization != nil && *src.TargetAverageUtilization < 1 {
        allErrs = append(allErrs, field.Invalid(fldPath.Child("targetAverageUtilization"), src.TargetAverageUtilization, "must be greater than 0"))
}
```
So from my point of view, -10 and -70 can be treated as the same and we can regard that to be duplicated.
Does it make sense? Feel free to tell me if I missed anything.

**Comment by allencloud** [comment]
Any update? @mbohlool 
Could we make this move on? And Does what I provide make sense?
Please feel free to tell me if I was something wrong.

**Comment by mbohlool** [comment]
yes. Sorry for late reply.

/lgtm

**Comment by allencloud** [comment]
And cloud someone comment an `approve no-issue`, since I think there is no related issue this PR refers to? ping @mbohlool @smarterclayton @sttts

**Comment by allencloud** [comment]
I think you have attached an `/approve no-issue`, while that is strange that this PR is till not approved. @mbohlool @smarterclayton

**Comment by xiangpengzhao** [comment]
@allencloud This is still not approved because @mbohlool is not an approver from `pkg/apis/OWNERS` though he commented `/approve no-issue`. You still need the exact comment from @smarterclayton or other approvers from `pkg/apis/OWNERS`.

**Comment by allencloud** [comment]
So, friendly ping @smarterclayton 🍻

**Comment by allencloud** [comment]
I am afraid here we need a kind approval to make this move on, as it seems to be blocked for a while. Could you help this? @smarterclayton @thockin @erictune @lavalamp @liggitt 
I am sorry for the ping and I tried to seek help from `pkg/apis/OWNERS`

**Pull request merged**